seatunnel-connectors-v2/connector-common/src/main/java/org/apache/seatunnel/connectors/seatunnel/common/source/arrow/reader/Arrow2SeatunnelRowReader.java

```diff
@@ -0,0 +1,264 @@
+import org.apache.seatunnel.shade.org.apache.arrow.vector.FieldVector;
+import org.apache.seatunnel.shade.org.apache.arrow.vector.VectorSchemaRoot;
+import org.apache.seatunnel.shade.org.apache.arrow.vector.ipc.ArrowStreamReader;
+import org.apache.seatunnel.shade.org.apache.arrow.vector.types.Types;
+import org.apache.seatunnel.shade.org.apache.arrow.vector.util.Text;
+
+import org.apache.seatunnel.api.table.type.ArrayType;
+import org.apache.seatunnel.api.table.type.MapType;
+import org.apache.seatunnel.api.table.type.SeaTunnelDataType;
+import org.apache.seatunnel.api.table.type.SeaTunnelRow;
+import org.apache.seatunnel.api.table.type.SeaTunnelRowType;
+import org.apache.seatunnel.api.table.type.SqlType;
+import org.apache.seatunnel.connectors.seatunnel.common.source.arrow.converter.Converter;
+
+import lombok.extern.slf4j.Slf4j;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+import java.util.function.Function;
+
+@Slf4j
+public class Arrow2SeatunnelRowReader implements AutoCloseable {
+
+    private final SeaTunnelDataType<?>[] seaTunnelDataTypes;
+    private int offsetInRowBatch = 0;
+    private int rowCountInOneBatch = 0;
+    private int readRowCount = 0;
+    private List<FieldVector> fieldVectors;
+    private VectorSchemaRoot root;
+    private final ArrowStreamReader arrowStreamReader;
+    private final RootAllocator rootAllocator;
+    private final Map<String, Integer> fieldIndexMap = new HashMap<>();
+    private final List<SeaTunnelRow> seatunnelRowBatch = new ArrayList<>();
+    private final List<Converter> converters = new ArrayList<>();
+    private final DateTimeFormatter DATE_FORMATTER = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+    private final DateTimeFormatter TIME_FORMATTER = DateTimeFormatter.ofPattern("HH:mm:ss");
+    private final DateTimeFormatter DATETIME_FORMATTER =
+            DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss");
+
+    public Arrow2SeatunnelRowReader(byte[] byteArray, SeaTunnelRowType seaTunnelRowType) {
+        this.seaTunnelDataTypes = seaTunnelRowType.getFieldTypes();
+        this.rootAllocator = new RootAllocator(Integer.MAX_VALUE);
+        this.arrowStreamReader =
+                new ArrowStreamReader(new ByteArrayInputStream(byteArray), rootAllocator);
+        for (int i = 0; i < seaTunnelRowType.getFieldNames().length; i++) {
+            fieldIndexMap.put(seaTunnelRowType.getFieldNames()[i], i);
+        }
+        ServiceLoader<Converter> load = ServiceLoader.load(Converter.class);
```

Comment:
why not use static load?

Reply:
![image](https://github.com/user-attachments/assets/47572270-9ae0-4ebc-b961-b3e969305c1b)
arrow currently has a total of 45 types, each type corresponds to a Vector, each Vector corresponds to a converter,  The SPI class loading method is a little more brief

Reply:
fixed